google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java

```diff
@@ -0,0 +1,174 @@
+        BigtableExporterUtils.getDefaultTaskValue());
+  }
+
+  @VisibleForTesting
+  BigtableCloudMonitoringExporter(
+      String projectId,
+      MetricServiceClient client,
+      MonitoredResource monitoredResource,
+      String taskId) {
+    this.client = client;
+    this.monitoredResource = monitoredResource;
+    this.taskId = taskId;
+    this.projectId = projectId;
+  }
+
+  @Override
+  public CompletableResultCode export(Collection<MetricData> collection) {
+    if (isShutdown.get()) {
+      logger.log(Level.WARNING, "Exporter is shutting down");
+      return CompletableResultCode.ofFailure();
+    }
+    if (!collection.stream()
+        .flatMap(metricData -> metricData.getData().getPoints().stream())
+        .allMatch(pd -> projectId.equals(BigtableExporterUtils.getProjectId(pd)))) {
+      logger.log(Level.WARNING, "Metric data has different a projectId. Skip exporting.");
+      return CompletableResultCode.ofFailure();
+    }
+
+    List<TimeSeries> allTimeSeries;
+    try {
+      allTimeSeries =
+          BigtableExporterUtils.convertCollectionToListOfTimeSeries(
+              collection, taskId, monitoredResource);
+    } catch (Throwable e) {
+      logger.log(Level.WARNING, "Failed to convert metric data to cloud monitoring timeseries.", e);
+      return CompletableResultCode.ofFailure();
+    }
+
+    ProjectName projectName = ProjectName.of(projectId);
+    CreateTimeSeriesRequest request =
+        CreateTimeSeriesRequest.newBuilder()
+            .setName(projectName.toString())
+            .addAllTimeSeries(allTimeSeries)
+            .build();
+
+    ApiFuture<Empty> future = this.client.createServiceTimeSeriesCallable().futureCall(request);
+
+    lastExportCode = new CompletableResultCode();
+
+    ApiFutures.addCallback(
+        future,
+        new ApiFutureCallback<Empty>() {
+          @Override
+          public void onFailure(Throwable throwable) {
+            logger.log(Level.WARNING, "createServiceTimeSeries request failed. ", throwable);
+            lastExportCode.fail();
+          }
+
+          @Override
+          public void onSuccess(Empty empty) {
+            lastExportCode.succeed();
+          }
+        },
+        MoreExecutors.directExecutor());
+
+    return lastExportCode;
+  }
+
+  @Override
+  public CompletableResultCode flush() {
+    if (lastExportCode != null) {
+      return lastExportCode;
+    }
+    return CompletableResultCode.ofSuccess();
+  }
+
+  @Override
+  public CompletableResultCode shutdown() {
+    if (!isShutdown.compareAndSet(false, true)) {
+      logger.log(Level.WARNING, "shutdown is called multiple times");
+      return CompletableResultCode.ofSuccess();
+    }
+    CompletableResultCode resultCode = flush();
+    try {
+      client.shutdown();
+    } catch (Throwable e) {
+      logger.log(Level.WARNING, "failed to shutdown the monitoring client", e);
+      return CompletableResultCode.ofFailure();
+    }
```

Comment:
you need to chain these:
```
CompletableResultCode closeFuture = new CompletableResultCode();
CompletableResultCode flushFuture = flush();

flushFuture.whenComplere(() -> {
    Exception closeError = null;
  try {
    client.shutdown();
  } catch(RuntimeException e) {
    closeError = e;
  }
  if (closeError == null) {
    closeFuture.succeed();
  } else {
    log....
   closeFuture.fail()
  } 
})
```

Otherwise you will close the client while its flushing